### PR TITLE
feat: rename blockchain connector type to application

### PR DESCRIPTION
**Pull request by jvallesm**
Because

- https://github.com/instill-ai/protobufs/pull/274 removed the Blockchain
  connector type and modified the connector and operator list view type.
- Once https://github.com/instill-ai/connector/pull/131 is merged, `numbers` connector will become an application connector.

This commit

- Updates the auto-generated proto Go code, adapting the backend to the changes:
  - Modifies the Blockchain enum value in the database via a migration.
  - Modifies type conversions to take Application connector into account.
  - Removes deprecated lookup methods.
  - Updates `VIEW_XXX` enum values.

### CODE REVIEW

File: pkg/worker/workflow.go [hunk 1]
```diff
@@ -309,39 +309,26 @@ func (w *worker) TriggerPipelineWorkflow(ctx workflow.Context, param *TriggerPip
...
 							return nil, err
 						}
-						recipePb := &structpb.Struct{}
-						err = protojson.Unmarshal(recipeByte, recipePb)
-						if err != nil {
-							return nil, err
-						}
-
-						// TODO: remove this hardcode injection
-						if comp.DefinitionName == "connector-definitions/70d8664a-d512-4517-a5e8-5d4da81756a7" {
```
Comment (jvallesm) [inline pkg/worker/workflow.go:326]: ℹ️ this was refactor as this `if` block was redundant.

File: pkg/worker/workflow.go [hunk 2]
```diff
@@ -309,39 +309,26 @@ func (w *worker) TriggerPipelineWorkflow(ctx workflow.Context, param *TriggerPip
...
+								"recipe": "${global.pipeline.recipe}",
+							},
+							"owner": map[string]interface{}{
+								"uid": "${global.owner.uid}",
+							},
+						})
 						if err != nil {
 							return nil, err
 						}
-						recipePb := &structpb.Struct{}
```
Comment (jvallesm) [inline pkg/worker/workflow.go:319]: ℹ️ `recipePb` was unused

File: pkg/db/migration/000011_init.down.sql [hunk 3]
```diff
@@ -0,0 +1,5 @@
+BEGIN;
+
+ALTER TYPE CONNECTOR_VALID_CONNECTOR_TYPE RENAME VALUE 'CONNECTOR_TYPE_APPLICATION' TO 'CONNECTOR_TYPE_BLOCKCHAIN';
```
Comment (jvallesm) [inline pkg/db/migration/000011_init.down.sql:3]: @donch1989 I didn't see any `down` migrations but still I wrote this one. Is there an issue with migrating down?

File: pkg/db/migration/000011_init.up.sql [hunk 4]
```diff
@@ -0,0 +1,5 @@
+BEGIN;
+
+ALTER TYPE CONNECTOR_VALID_CONNECTOR_TYPE RENAME VALUE 'CONNECTOR_TYPE_BLOCKCHAIN' TO 'CONNECTOR_TYPE_APPLICATION';
```
Comment (jvallesm) [inline pkg/db/migration/000011_init.up.sql:3]: @donch1989 I ran the migration command locally and all the references to this type in other tables (i.e. `connector`) were updated successfully.

I'm not aware how we're conducting migrations at this moment in staging / production, let's keep this in mind for the next release.

**Comment by jvallesm** [comment]
## ✅ QA

- On `main`, I created a `Numbers` connector and a pipeline using it
- I switched to `jvalles/ins-3739-rename-blockhain-type-to-application` in `pipeline-backend`, `connector` and `console`.
- The pipeline kept working ✅ 
- A new pipeline with a new connector worked as well ✅ 

Additionally, I tested the API endpoint:

```sh
$ curl --request GET \
  --url 'localhost:8080/vdp/v1beta/connector-definitions?view=VIEW_BASIC&filter=connector_type=CONNECTOR_TYPE_APPLICATION' \
  --header 'accept: application/json' \
  --header 'Authorization: Bearer insXXX' | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   699  100   699    0     0  16364      0 --:--:-- --:--:-- --:--:-- 16642
{
  "connector_definitions": [
    {
      "name": "connector-definitions/numbers",
      "uid": "70d8664a-d512-4517-a5e8-5d4da81756a7",
      "id": "numbers",
      "title": "Numbers Protocol",
      "documentation_url": "https://www.instill.tech/docs/latest/vdp/app-connectors/numbers-protocol",
      "icon": "assets/numbers.svg",
      "spec": null,
      "type": "CONNECTOR_TYPE_APPLICATION",
      "tombstone": false,
      "public": true,
      "custom": false,
      "vendor": "Numbers Protocol",
      "vendor_attributes": null,
      "source_url": "https://github.com/instill-ai/connector/blob/main/pkg/numbers/v0",
      "version": "0.1.0-alpha",
      "tasks": [
        {
          "name": "TASK_REGISTER",
          "title": "Register",
          "description": "Register a file into NumbersProtocol Chain."
        }
      ]
    }
  ],
  "next_page_token": "",
  "total_size": 1
}
$ curl --request GET \
  --url 'localhost:8080/vdp/v1beta/connector-definitions?view=VIEW_BASIC&filter=connector_type=CONNECTOR_TYPE_BLOCKCHAIN' \
  --header 'accept: application/json' \
  --header 'Authorization: Bearer insXXX' | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   104  100   104    0     0   7888      0 --:--:-- --:--:-- --:--:--  8000
{
  "code": 2,
  "message": "check call expr: undeclared identifier 'CONNECTOR_TYPE_BLOCKCHAIN'",
  "details": []
}
```

File: pkg/db/migration/000011_init.down.sql (hunk 3 above)
Comment (donch1989) [reply pkg/db/migration/000011_init.down.sql:3]: For us, the migration down is not essential for now, but I recall that the library will raise an error if we don't provide the down file. Therefore, we simply create empty ones.

File: pkg/db/migration/000011_init.up.sql (hunk 4 above)
Comment (donch1989) [reply pkg/db/migration/000011_init.up.sql:3]: The staging/production use the same command `pipeline-backend-migrate`. So if the Core works, then the cluster should work as well.

**Review by donch1989 — APPROVED** [review]

**Pull request merged**